### PR TITLE
Migration to Tensorflow 2.0

### PR DESCRIPTION
**Pull request by lojzezust**
Code was refactored according to Tensorflow 2.0 guidelines. Models are now defined as Keras models. This should enable easier model building and cleaner code. FCN and TFCN models have been updated to fit.

Most of the features of 1.0 have been ported to 2.0.

List of implemented features:
* `FCNModel`, `TFCNModel` implementations as Keras models
* `SegmentationModel`: abstract base for all segmentation models. Contains loss and metric functions. Custom losses for segmentation should go here. Takes care of cropping logits for loss and metric computation.
* `train` and `train_and_evaluate` methods. These are wrappers over the keras `Model.fit` method. They take care of model checkpointing and summary saving to tensorboard.
* `fit`, `evaluate`, `predict` keras methods can be also used on the model. Since the model inherits from keras model these methods can take all types of inputs (from numpy arrays to tf.data datasets)
* Other keras methods can also be used for saving the model, etc.
* Tasks have been updated to work with the new model with minimal configuration file changes

Missing features (not yet implemented):
* RFCNModel
* Exporting custom summaries to tensorboard

### CODE REVIEW

**Comment by devisperessutti** [comment]
@lojzezust I have made minor changes to the code for formatting compatibility to PEP8, and to fix the example model. I am happy to merge if you are happy with my changes
